src/Microsoft.AspNetCore.Cors/Infrastructure/UriExtensions.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Cors.Infrastructure
+{
+    internal static class UriExtensions
+    {
+        public static bool IsSubdomainOf(this Uri subdomain, Uri domain)
+        {
+            return subdomain.IsAbsoluteUri 
+                && domain.IsAbsoluteUri
+                && subdomain.Scheme == domain.Scheme
+                && subdomain.Host.EndsWith(domain.Host);
```

Comment:
1. I don't think this check is right: `"example.com".EndsWith("ample.com") == true`. Does this logic need to be a bit smarter about being a "subdomain"?
2. Does this also need to check things like ports? Is that a factor in CORS calculation?